src/boxes/sampleentries/sampleentry.ts

```diff
@@ -25,8 +25,8 @@ function decimalToHex(d: number | string, padding?: number | null) {
 }
 
 class avcCSampleEntryBase extends VisualSampleEntry {
-  declare avcC: avcCBox;
-  declare avcCs: Array<avcCBox>;
+  avcC: avcCBox | undefined;
```

Comment:
unsure about this

Reply:
My reasoning is that the avcC is at some point undefined (same for all other boxes) since we don't call parse in the CTor

Reply:
I'm not sure why no `| undefined` typings are working?

![image](https://github.com/user-attachments/assets/aacb05b5-673c-4742-a344-22ff4e1bda85)
![Screenshot 2025-05-28 at 11 09 00](https://github.com/user-attachments/assets/b0845595-d1b7-47b5-84e0-d740480f1ced)

On the first screen Typescript doesn't even pick the `undefined` in the second screen Typescript allows to use .length on an optional type
I'm clueless 🤔

Reply:
I'm thinking of dropping the latest commit and current work (unmerged) to merge this PR and resolve issues, then come back to it

Reply:
> I'm not sure why no | undefined typings are working?

This is probably due to `strict: false` in the  tsconfig. Maybe not a bad idea to set it to true?